core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -302,6 +302,53 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  def setProjectSpendConfiguration(projectName: RawlsBillingProjectName, datasetName: String): Future[Int] = {
+
+    validateBigQueryDatasetName(datasetName)
+
+    requireProjectAction(projectName, SamBillingProjectActions.alterSpendReportConfiguration) {
+      val bqService = bqServiceFactory.getServiceFromCredentialPath(pathToCredentialJson, GoogleProject(projectName.value))
+
+      for {
+        //Get the dataset to validate that it exists and that we have permission to see it
+        _ <- bqService.use(_.getDataset(datasetName)).unsafeToFuture().map {
+          case None => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The dataset $datasetName could not be found."))
+          case dataset => dataset
+        }
+
+        //The billing account ID in the DB can't be trusted, so we'll get it directly from the source of truth
+        billingAccountIdOpt <- gcsDAO.getProjectBillingAccount(projectName, userInfo)
+
+        //Get the parent billing account for the projects billing account
+        parentBillingAccountId <- billingAccountIdOpt match {
```

Comment:
```suggestion
        rootBillingAccountId <- billingAccountIdOpt match {
```
maybe rename this? since it is looking for the top-most level (which may be a parent or not)

Reply:
good call, I was struggling to come up with a good word to use here but that's definitely better!